lib/eventsource.js

```diff
@@ -46,50 +50,35 @@ function EventSource (url, eventSourceInitDict) {
   var self = this
   self.reconnectInterval = 1000
 
-  function onConnectionClosed (message) {
-    if (readyState === EventSource.CLOSED) return
-    readyState = EventSource.CONNECTING
-    _emit('error', new Event('error', {message: message}))
-
-    // The url may have been changed by a temporary
-    // redirect. If that's the case, revert it now.
-    if (reconnectUrl) {
-      url = reconnectUrl
-      reconnectUrl = null
-    }
-    setTimeout(function () {
-      if (readyState !== EventSource.CONNECTING) {
-        return
-      }
-      connect()
-    }, self.reconnectInterval)
-  }
-
   var req
   var lastEventId = ''
-  if (eventSourceInitDict && eventSourceInitDict.headers && eventSourceInitDict.headers['Last-Event-ID']) {
-    lastEventId = eventSourceInitDict.headers['Last-Event-ID']
-    delete eventSourceInitDict.headers['Last-Event-ID']
+  if (config.headers && config.headers['Last-Event-ID']) {
+    lastEventId = config.headers['Last-Event-ID']
   }
 
   var discardTrailingNewline = false
   var data = ''
   var eventName = ''
 
   var reconnectUrl = null
-
-  function connect () {
+  var retryDelayStrategy = new retryDelay.RetryDelayStrategy(
+    config.initialRetryDelayMillis !== null && config.initialRetryDelayMillis !== undefined ? config.initialRetryDelayMillis : 1000,
```

Comment:
Minor: isn't this equivalent to `config.initialRetryDelayMills || 1000`?

Reply:
No. They could set it explicitly to zero, in which case it would be falsey.

Reply:
Aha, indeed.